asserts/snap_asserts.go

```diff
@@ -110,6 +117,8 @@ func (snapdcl *SnapDeclaration) Prerequisites() []*Ref {
 	}
 }
 
+var validAppCommand = regexp.MustCompile("^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$")
```

Comment:
@mvo5 @niemeyer  that's a good question, but that's the same regexp we use for apps, if we want to do that we need to support mapping from one to the other or we need to relax the one for apps as well